File: api/NAGP.Api.Data/DataModels/Movie.cs

namespace NAGP.Api.Data.DataModels;

public class Movie
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string? Synopsis { get; set; }

    public DateTimeOffset ReleaseDate { get; set; }

    public string? ImageUrl { get; set; }

    public int Rating { get; set; }
}

File: app/src/lib/models/movie.ts
export type Movie = {
    id: number;
    name: string;
    releaseDate: string;
    synopsis: string;
    rating: number;
};

export function parseMovie(data: Movie) {
    return {
        ...data,
        releaseDate: new Date(data.releaseDate),
        releaseDateStr: new Date(data.releaseDate).toDateString(),
        releaseYear: new Date(data.releaseDate).getFullYear(),
        rating: parseInt(data.rating.toString()) // Ensure rating is a float with one decimal
    };
}